eahub/profiles/tests/test_views.py

```diff
@@ -0,0 +1,41 @@
+from django.test import override_settings
+
+from eahub.tests.cases import EAHubTestCase
+from eahub.profiles.models import VisibilityEnum
+
+
+@override_settings(STATICFILES_STORAGE='django.contrib.staticfiles.storage.StaticFilesStorage')
+class ProfileTestCase(EAHubTestCase):
+
+    def test_private_profile_hidden(self):
+        profile = self.gen.profile(first_name="first", last_name="last")
+        profile.visibility = VisibilityEnum.PRIVATE
+        profile.save()
+        response = self.client.get('/profile/first-last/')
+        self.assertEqual(response.status_code, 404)
+
+    def test_internal_profile_hidden(self):
+        profile = self.gen.profile(first_name="first", last_name="last")
+        profile.visibility = VisibilityEnum.INTERNAL
+        profile.save()
+        response = self.client.get('/profile/first-last/')
```

Comment:
FYI optional, but could also be reversed, eg as `reverse("profiles_app:profile", {"slug"=profile.slug})`

Reply:
I also sometimes just create a method on the model as `def get_url(self) -> str`

Reply:
done